### PR TITLE
[portsorch]: Implement port PFC asym capability check

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@nvidia.com>

Capability check helps to avoid port misconfiguration on systems where PFC asymmetric is not supported

**What I did**
* Added PFC asymmetric configuration capability check

**Why I did it**
* To avoid port configuration failures

**How I verified it**
1. Run VS tests

**Details if related**
* N/A

### CODE REVIEW

**Review by Yakiv-Huryk — APPROVED** [review]

**Comment by nazariig** [comment]
@prsunny / @qiluo-msft please help to review & merge

**Comment by nazariig** [comment]
@prsunny / @qiluo-msft just a kind reminder

**Comment by nazariig** [comment]
@prsunny / @qiluo-msft just a kind reminder

File: orchagent/port/portcap.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// includes -----------------------------------------------------------------------------------------------------------
```
Comment (prsunny) [inline orchagent/port/portcap.cpp:1]: Can we please rename to `portcapblty.cpp` or similar? `portcap `is slightly misleading.

**Review by dgsudharsan — APPROVED** [review]

**Review by volodymyrsamotiy — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@StormLiangMS  Tested on top of 202305 commit https://github.com/sonic-net/sonic-buildimage/commit/40d1159b003eb73e3408ea14926d0f2c7b8790c9

**Comment by StormLiangMS** [comment]
hi @dgsudharsan this is an enhancement, let's have it in next release.

**Comment by dgsudharsan** [comment]
@StormLiangMS  This is not enhancement but a bug fix. If a platform doesn’t support pfc asym, by default SONiC will crash. We  added a capability to check if pfc asym is supported before programming the configuration to SAI

**Pull request merged**